package/src/components/MessageInput/MessageInput.tsx

```diff
@@ -281,7 +285,19 @@ const MessageInputWithContext = <
       );
       return !uploadedImage;
     });
-    imagesToUpload.forEach((image) => uploadNewImage(image));
+    // Check if the file size of the image exceeds the threshold of 100MB
+    if (
+      maxFileSizeToUploadInMb &&
+      imagesToUpload[0].fileSize &&
+      Number(imagesToUpload[0].fileSize) / MegaByesToBytes > maxFileSizeToUploadInMb
```

Comment:
What is the idea behind checking the fileSize only of the first one in the array?

Reply:
This is due to bad implementation; the work can be done by a `find` operator too, but I don't know why a `filter` is used. I didn't find a case where there would be multiple files here, and the array length will always be one. Since I didn't touch the implementation, I used the filtered first array element here hence only `0` is used.

Reply:
Changed it to `find`.